platform/linux-generic/odp_random_null.c

```diff
@@ -24,12 +29,14 @@ static int32_t _random_data(uint8_t *buf, uint32_t len, uint32_t *seed)
 		uint32_t rand_word;
 		uint8_t rand_byte[4];
 	} u;
-	uint32_t i = 0, j;
+	uint32_t i = 0, j, k;
 
 	while (i < len) {
 		u.rand_word = rand_r(seed);
 
-		for (j = 0; j < 4 && i < len; j++, i++)
+		/* Use two least significant bytes */
```

Comment:
This actually requires that RAND_MAX is a power of two minus 1. Otherwise you get a biased distribution and may see the original problem in some form. In practice RAND_MAX satisfies the requirement in a typical Linux distro, so maybe the easiest fix is to just additionally assert that (RAND_MAX & (RAND_MAX + 1)) == 0. If you did not want to do that, you would probably have to discard rand_r results that exceed the highest power-of-two-minus-one that is less or equal to RAND_MAX and then retry rand_r. But maybe it is not worth the trouble for this piece of code.

You might also want to tune the commit message a little (it talks about two "first" bytes). And no need for the suggested-by tag.

Reply:
Fixed in V3